bliss/simulator/simulated_dataset.py

```diff
@@ -87,3 +89,77 @@ def val_dataloader(self):
 
     def test_dataloader(self):
         return DataLoader(self, batch_size=None, num_workers=self.num_workers)
+
+
+class CachedSimulatedDataset(pl.LightningDataModule, Dataset):
+    def __init__(
+        self,
+        simulator: SimulatedDataset,
```

Comment:
Better to just pass `batch_size` to the constructor, rather than passing it the whole simulator. We may want to use a different `batch_size` when simulating data than when loading cached data. And anyway, it seems cleaner to decouple `CachedSimulatedDataset` from `SimulatedDataset`

Reply:
Done.